samples/snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsModifiers/CS/csrefKeywordsModifiers.cs

```diff
@@ -798,4 +798,49 @@ public override string Name
         }
     }
     //</snippet26>
+
+    public class AbstractExercise3
+    {
+        //<snippet27>
+        public abstract class Shape
+        {
+            public string Color { get; set; }
+
+            // Constructor of the abstract class
+            protected Shape(string color)
+            {
+                Color = color;
+                Console.WriteLine("Created a shape with color {color}.");
+            }
+
+            // Abstract method that must be implemented by derived classes
+            public abstract double CalculateArea();
+        }
+
+        public class Square : Shape
+        {
+            public double Side { get; set; }
+
+            // Constructor of the derived class calling the base class constructor
+            public Square(string color, double side) : base(color)
+            {
+                Side = side;
+            }
+
+            public override double CalculateArea()
+            {
+                return Side * Side;
+            }
+        }
+
+        public class Program
+        {
+            public static void Main(string[] args)
+             {
+                    Square square = new Square("red", 5);
+                    Console.WriteLine("Area of the square: {square.CalculateArea()}");            
+             }
+        }
+        //</snippet26>
```

Comment:
This needs to match the opening snippet:

```suggestion
        //</snippet27>
```